### PR TITLE
Filex to open files

### LINKED ISSUE #1663: Can not  view files/pdf's in acter chat
there are several options to share, but not to view or download.

Image

### PR DESCRIPTION
**Pull request by gnunicorn**
built on top of #2048  this adds facilities to open or share files.

Specifically, this replaces the existing `DownloadButton` with a new `ShareFileButton` (from the newly added `files` feature) that will open a bottom up drawer for the file in question and allows the user to chose between
 1. opening the file (with anything the system provides for it), 
 2. share the file (via the system provided sharing facilities), or
 3. save it to disk (not supported on Android because of #1803 ).

See it in action:

https://github.com/user-attachments/assets/2252d5be-1a33-4bfe-b661-6d43635a8dd1

Fixes #1663.
Additionally this adds the Phosphor-Icons library as it had a nicer icon and we wanted to start using that anyways...

### CODE REVIEW

File: app/pubspec.yaml [hunk 1]
```diff
@@ -133,6 +133,7 @@ dependencies:
   shake_detector:
     path: ../packages/shake_detector
   open_filex: ^4.5.0
+  phosphor_flutter: ^2.1.0
```
Comment (kumarpalsinh25) [inline app/pubspec.yaml:136]: In recent past, I have explore this package and check with Acter app. Here is the my findings.

Advantages:
- Phosphor have better look and feel compare to Atlas
- Phosphor have more number of icons compare to Atlas

Disadvantages:
- Phosphor have less variety for single icon category

**Review by kumarpalsinh25 — APPROVED** [review]
Useful addition to File management.

File: app/lib/features/files/actions/file_share.dart [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      padding: const EdgeInsets.symmetric(horizontal: 20.0),
+      constraints: const BoxConstraints(
+        maxWidth: 600,
+        minWidth: 300,
+      ),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          header ?? const SizedBox.shrink(),
+          if (header != null) const SizedBox(height: 16.0),
```
[the remark this answers is not in the document]
Comment (gnunicorn) [reply app/lib/features/files/actions/file_share.dart:62]: if there is a header, we want some padding between it and the items.

File: app/lib/features/files/actions/file_share.dart [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    ),
+  );
+}
+
+class _FileOptionsDialog extends StatelessWidget {
+  final File file;
+  final String? mimeType;
+  final Widget? header;
+  final List<Widget>? beforeOptions;
+  final List<Widget>? afterOptions;
```
[the remark this answers is not in the document]
Comment (gnunicorn) [reply app/lib/features/files/actions/file_share.dart:39]: like what? it allows you to inject further options in the list... either at the top or bottom. I ended up not using it, I can also remove it...

File: app/lib/features/files/actions/file_share.dart [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                // done, close this dialog
+                if (context.mounted) {
+                  Navigator.pop(context);
+                }
+              }
+            },
+            label: Text(lang.shareFile),
+            icon: PhosphorIcon(PhosphorIcons.shareNetwork()),
+          ),
+          if (!Platform.isAndroid) // crashes on Android for some reason ...
```
[the remark this answers is not in the document]
Comment (gnunicorn) [reply app/lib/features/files/actions/file_share.dart:91]: possibly. if you have an idea for a fix, the bug is reported here https://github.com/acterglobal/a3/issues/1803

File: app/pubspec.yaml (hunk 1 above)
Comment (gnunicorn) [reply app/pubspec.yaml:136]: >     Phosphor have less variety for single icon category

I don't understand what that means. Can you elaborate?

File: app/lib/features/events/pages/event_details_page.dart [hunk 5]
```diff
@@ -419,51 +420,29 @@ class _EventDetailPageConsumerState extends ConsumerState<EventDetailPage> {
...
-          ),
-        ),
-      ],
+    return IconButton(
+      icon: PhosphorIcon(PhosphorIcons.shareFat()),
+      onPressed: () => onShareEvent(calendarEvent),
     );
   }
 
   Future<void> onShareEvent(CalendarEvent event) async {
```
[the remark this answers is not in the document]
Comment (gnunicorn) [reply app/lib/features/events/pages/event_details_page.dart:429]: could do, but it is single use case ... so I'll keep it here for now.

File: app/pubspec.yaml (hunk 1 above)
Comment (kumarpalsinh25) [reply app/pubspec.yaml:136]: Lets example of icons with `Email` category. Atlas have more variety in single category.

Icons

Comment (gnunicorn) [reply app/pubspec.yaml:136]: ah. gotcha ... not sure that matter too much though...

Comment (kumarpalsinh25) [reply app/pubspec.yaml:136]: https://github.com/user-attachments/assets/28e670b7-a1aa-46a3-a8e2-e9c4c7ddaeee

File: app/lib/features/files/actions/file_share.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (kumarpalsinh25) [reply app/lib/features/files/actions/file_share.dart:91]: If it is the case what I am thinking then it require bit of discussion on the same. I will cross check details and update you.

**Pull request merged**